operator-common/src/main/java/io/strimzi/operator/common/Operator.java

```diff
@@ -61,6 +61,8 @@ default void reconcileAll(String trigger, String namespace, Handler<AsyncResult<
                 reconcileThese(trigger, ar.result(), namespace, handler);
                 periodicReconciliationsCounter(namespace).increment();
             } else {
+                resourceCounter(namespace).set(0);
+                pausedResourceCounter(namespace).set(0);
```

Comment:
So, how does this work with:
* Failures to all-namespaces operator? It will IMHO at any failure set the metrics for namespace `*` to 0 and never remove them. And at the same time it will keep the old metrics for namespaces which were maybe deleted.
* How do you know the count is 0 because the namespace was deleted? How do you know it wasn't just some temp glitch in Kube API or RBAC issue but exactly deleted namespace? You should probably remove the metrics rarther than setting them to 0.

I think that by fixing one very edge case you break several other very edge cases? I really think that we should accept that when the operator is broken due to things such as RBAC issues or missing namespaces etc., it will not provide reliable metrics since it does not really know the actual values. Otherwise you might take forever fixing all weird cases.

CC @ppatierno 

_As a sidenote => in few months, nobody will remember what was this piece of code do or the test you added test. You should definitely explain in places like this in comments why are they there._

Reply:
I would agree with Jakub on this. Handling the scenario here means catching not just the case I raised but even something more ... and the cases could be a lot. 
I thought more about it:

* the metrics seem to be reset when the CO is configured to watch ALL namespaces but you delete one. So it's fine.
* if you instead configure the CO to watch specific namespaces but then delete one you are kind of breaking CO configuration so I agree with Jakub, we have just to accept it.

In the end we could just close the issue with no fix.

Reply:
Yes, I agree with closing this. Should the CO at least print some warnings if the code flow goes thtough https://github.com/strimzi/strimzi-kafka-operator/blob/main/operator-common/src/main/java/io/strimzi/operator/common/Operator.java#L64 ?

Reply:
I would expect the errors to be logged on other level. But you would need to double check it if you want.